tests/Unit/Time/Test_Slab.cpp

```diff
@@ -0,0 +1,277 @@
+
+TEST_CASE("Unit.Time.Slab", "[Unit][Time]") {
+  Approx approx =
+      Approx::custom().epsilon(5 * std::numeric_limits<double>::epsilon());
+
+  const double tstart_d = 0.68138945475734402635;
+  const double tend_d = 76.34481744714527451379;
+  // Make sure we're using values that will trigger rounding errors.
+  CHECK_FALSE(tstart_d + (tend_d - tstart_d) == tend_d);
+  const Slab slab(tstart_d, tend_d);
+
+  // Arbitrary
+  const double tend2_d = tend_d + 1.234;
+
+  {
+    CHECK(slab.start().value() == tstart_d);
+    CHECK(slab.end().value() == tend_d);
+    CHECK(slab.duration().value() == approx(tend_d - tstart_d));
+    CHECK(slab.end() - slab.start() == slab.duration());
+  }
+
+  {
+    const Slab slab2 = Slab::new_with_duration(tstart_d, tend_d);
+    CHECK(slab2.start().value() == tstart_d);
+    CHECK(slab2.duration().value() == approx(tend_d));
+  }
+
+  {
+    const Slab slab2 = Slab::new_with_duration_reverse(tend_d, tstart_d);
+    CHECK(slab2.end().value() == tend_d);
+    CHECK(slab2.duration().value() == approx.scale(tend_d)(tstart_d));
+  }
+
+  {
+    const Slab next = slab.advance();
+    CHECK(next.start() == slab.end());
+    CHECK(next.duration().value() == approx(slab.duration().value()));
+    const Slab prev = slab.retreat();
+    CHECK(prev.end() == slab.start());
+    CHECK(prev.duration().value() == approx(slab.duration().value()));
+
+    CHECK(slab.advance_towards(slab.duration()) == next);
+    CHECK(slab.advance_towards(-slab.duration()) == prev);
+  }
+
+  {
+    const Slab front = slab.with_duration(tend2_d);
+    CHECK(front.start().value() == slab.start().value());
+    CHECK(front.duration().value() == approx(tend2_d));
+    const Slab back = slab.with_duration_reverse(tend2_d);
+    CHECK(back.end().value() == slab.end().value());
+    CHECK(back.duration().value() == approx(tend2_d));
+  }
+
+  {
+    CHECK(slab.is_followed_by(slab.advance().with_duration(3)));
+    CHECK(slab.is_preceeded_by(slab.retreat().with_duration_reverse(3)));
+  }
+
+  CHECK(slab == slab);
+  CHECK_FALSE(slab == Slab(tstart_d, tend2_d));
+  CHECK(slab != Slab(tstart_d, tend2_d));
+  CHECK(slab != Slab(tend2_d / 2., tend_d));
+  CHECK(slab != Slab(tend_d, tend2_d));
+
+  check_cmp(Slab(1, 2), Slab(3, 4));
+  check_cmp(Slab(1, 2), Slab(2, 4));
+
+  // Hashing
+  std::hash<Slab> h;
+  CHECK(h(Slab(0, 1)) != h(Slab(0, 2)));
+  CHECK(h(Slab(0, 1)) != h(Slab(-1, 0)));
+  CHECK(h(Slab(0, 1)) != h(Slab(-1, 1)));
+
+  // Output
+  CHECK(get_output(Slab(0.5, 1.5)) == "Slab[0.5,1.5]");
+}
+
+TEST_CASE("Unit.Time.Slab.serialization", "[Unit][Time][Serialization]") {
+  const Slab slab(1.7, 2.4);
+  CHECK(slab == serialize_and_deserialize(slab));
+}
+
+// Failure tests
+#ifdef __clang__
+#pragma GCC diagnostic ignored "-Wunused-comparison"
+#endif
+
+// [[OutputRegex, Backwards Slab]]
+[[noreturn]] TEST_CASE("Unit.Time.Slab.Backwards.0", "[Unit][Time]") {
+  ASSERTION_TEST();
+#ifdef SPECTRE_DEBUG
+  Slab(1., 0.);
+  ERROR("Bad end");
```

Comment:
The current form is mandated by our dev guide.  Please change it there first if it is going to be changed.